src/logplex_tls.erl

```diff
@@ -83,25 +131,38 @@ verify_none(_, valid, UserState) ->
 verify_none(_, valid_peer, UserState) ->
     {valid, UserState}.
 
-verify_host(_Cert, {bad_cert, _}=Reason, UserState) ->
-    log_error(Reason, UserState),
+verify_host(_Cert, {bad_cert, Reason}, UserState) ->
+    ?ERR("channel_id=~p drain_id=~p dest=~s depth=~b at=verify_host "
+         "err=bad_cert reason=~p",
+         log_args(UserState, [Reason])),
     {fail, Reason};
-verify_host(_Cert, {extension, _}=Reason, UserState) ->
-    {unknown, UserState};
-verify_host(_Cert, valid, UserState) ->
-    {valid, UserState};
+verify_host(_Cert, {extension, _}, UserState0) ->
+    {unknown, UserState0};
+verify_host(_Cert, valid, UserState0) ->
+    {valid, incr_depth(UserState0)};
 verify_host(Cert, valid_peer, #user_state{ host=Host }=UserState) ->
```

Comment:
Not sure why the `incr_depth` is happening here, and at the other place (`handle_reply`).